### DISCUSSION TITLE
Performance benefits from React Compiler

### DISCUSSION THREAD

**Discussion by imteekay** [reduxjs/react-redux#2175] [General]
With the release of the [experimental React Compiler](https://react.dev/learn/react-compiler) in [React 19](https://react.dev/blog/2024/04/25/react-19), I was wondering how much a React app that heavily relies on Redux will be benefited (thinking about performance).

- For the cases we use `useState` / local state, there's a clear perf benefit
- For all selectors, redux store, etc, would it influence the app's performance in any way?

**Comment by phryneas** [comment]
It would just have the same impact as it would have with `useState`, it isn't doing any specific optimizations based on what kind of state management you are using.

**Reply by imteekay** [reply]
So using the new compiler _could_ be very beneficial (thinking about perf) for apps that rely on redux, right? 
In the future, we wouldn't need to cache with reselect anymore or we would still need it because they do different things/optimizations?

**Reply by phryneas** [reply]
Reselect prevents React component rerenders, the compiler minimizes the amount of children that rerender and the amount of computations inside a component.

Reselect also keeps return values more stable, which will then again help the compiler do it's job. They go hand in hand.
